IFC_Engine/Modify/SetIfcProperty.cs

```diff
@@ -0,0 +1,117 @@
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.oM.Adapters.IFC.Properties;
+using BH.oM.Base;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Adapters.IFC
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Attaches property to a BHoM object, which will be applied to a correspondent IFC element on Push.")]
+        [Input("bHoMObject", "BHoMObject to which the property will be attached.")]
+        [Input("name", "Name of the property to be attached.")]
+        [Input("value", "Value of the property to be attached.")]
+        [Output("bHoMObject")]
+        public static IBHoMObject SetIfcProperty(this IBHoMObject bHoMObject, string propName, object value)
+        {
+            if (bHoMObject == null)
+                return null;
+
+            List<IfcProperty> properties = new List<IfcProperty> { new IfcProperty { Name = propName, Value = value } };
+
+            IfcPropertiesToPush existingFragment = bHoMObject.Fragments.FirstOrDefault(x => x is IfcPropertiesToPush) as IfcPropertiesToPush;
+            if (existingFragment != null)
+            {
+                foreach (IfcProperty prop in existingFragment.Properties)
+                {
+                    if (prop.Name != propName)
+                        properties.Add(prop);
+                }
+            }
+
+            IfcPropertiesToPush fragment = new IfcPropertiesToPush { Properties = properties };
+
+            IBHoMObject obj = bHoMObject.ShallowClone();
+            obj.Fragments = new FragmentSet(bHoMObject.Fragments.Where(x => !(x is IfcPropertiesToPush)).ToList());
+            obj.Fragments.Add(fragment);
```

Comment:
This can be simplified: 
```suggestion
            bHoMObject.Fragments.AddOrReplace(fragment);
```

Reply:
Agreed with the suggestion to use `AddOrReplace`, although I think I still need a call to `ShallowClone` to avoid upstream changes? The latter would not be needed if the method returned `void` IMO. Or am I missing a point?

Reply:
Yes sorry my bad. Either do shallowclone first or use `BH.Engine.Base.AddFragment` with `replace` set to `true`.